### DISCUSSION TITLE
2.1 dreambooth  character training settings that worked for me

### DISCUSSION THREAD

**Discussion by zahdab** [d8ahazard/sd_dreambooth_extension#574] [Show and tell]
It seems dreambooth training on 2.1 affects the model way more drastically than 1.5. i tried following the same steps usually training 40-70 images, 100 steps per image with description,  with pre-generated class images but it yielded very bad results the model itself became very unusable.

Here are things i changed:
i reduced the number of the subject images to 20 with text description of each image something like "a closeup photo of [subject] person, sitting on a chair, with hands on hair, living room background", i reduced the learning rate to 0.0000015, trained for 100 steps per image. I trained the 768 model with 512 resolution images as i couldn't get it to train with 768 resolution images, change the setting in dreambooth to train with 512 size images.  generated class images via dreambooth. used all the usual settings to ensure memory usage is reduced except the ones that affect quality.  i did **set memory attention to xformers** as without that the class images seemed to be all blank. 

The subject seemed to not have the full resemblance of the trained person, so i retrained for an additional 30 steps per image at learning rate  0.000003. 

one thing to note is the subject i trained was a lebanese celebrity so sd did have a slight idea of what she looked like but was quite bad.. so not sure if this affects the amount of training needed for other subjects but maybe the old trick of picking the name of someone that resembles the subject helps.

the end results was not bad, a bit less flexible than the standard 2.1 model but still flexible enough to produce good quality images. 

Hopefully this helps someone as i've been struggling to trying to get 2.1 dreambooth training to work, if someone has better set of params please do share.  

results can be found here:
https://imgur.com/a/PH2uvql

**Comment by poondoggle** [comment]
Thanks for sharing! Can you post your "set COMMANDLINE_ARGS=" parameters as well as your Parameters > Advanced > Tuning in Dreambooth? I have an RTX3060 with 12GB and can't check the "Train Text Encoder" with a 2.1 model at all and if that isn't checked the results are really bad.

**Comment by zahdab** [comment]
set COMMANDLINE_ARGS= --xformers  is the only one i have

in advanced i check the 
8bit adam --> mixed precision fp16
use ema
memory attention --> xformers 

i have 3080 ti on a laptop which has 16gb vram  

also if training with a 768 model,, i still used 512 images and changed the setting to specify that
